pkg/arvo/app/chat-hook.hoon

```diff
@@ -366,8 +366,11 @@
       =/  chat-path  [%mailbox path.act]
       :_  state
       [%pass chat-path %agent [ship.act %chat-hook] %watch chat-path]~
-    ::  TODO: only ask for backlog from previous point
-    =/  chat-history  [%backlog (weld path.act /0)]
+    =/  mailbox=(unit mailbox)  (chat-scry path.act)
+    =/  chat-history=path
+      :-  %backlog
+      %+  weld  path.act
+      ?~(mailbox /0 /(scot %ud (lent envelopes.u.mailbox)))
```

Comment:
I think you explicitly want the sequence number registered with the last-heard message, instead of the length? If I hadn't previously gotten history, or never got full history, or am not entirely in sync history-wise for any other reason, then `lent` here will differ from the one on the host side.

Reply:
lent would be strictly less than or equal to the number of messages in the channel, so I’m not concerned. Worst case is a small number of duplicated messages, which is much better than previous behavior.

Reply:
> lent would be strictly less than or equal to the number of messages in the channel

I've now observed this not holding true for channels which get deleted, then recreated under the same name/path. It's making me miss out on messages whenever a reconnect happens. Using latest-heard message number here is more correct. (Doesn't fix the load-backlog-on-join case though.)